.github/workflows/dev-build.yml

```diff
@@ -71,5 +71,35 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - uses: actions/upload-artifact@v1
         with:
-          name: artifacts
+          name: MaaBundle-x64-Windows
           path: artifacts
+
+  macos-latest:
+    name: macos-latest
+    runs-on: macos-12
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Cache Homebrew
+        uses: actions/cache@v3
+        with:
+          path: $(brew --prefix)
+          key: ${{ runner.os }}-homebrew-${{ hashFiles('.config/brew/Brewfile') }}
+      - name: Install Dependencies
+        run: |
+            brew update --preinstall
+            brew install ninja range-v3
+      - name: Configure MeoAssistant
+        run: |
+          mkdir build && cd build
+          cmake .. -GNinja -DBUILD_XCFRAMEWORK=ON -DBUILD_UNIVERSAL=ON
+      - name: Build libMeoAssistant
+        run: cmake --build build
+      - name: Build MeoAsstMac
+        working-directory: src/MeoAsstMac
+        run: xcodebuild ARCHS="arm64 x86_64" CODE_SIGN_IDENTITY="-" DEVELOPMENT_TEAM="-" ONLY_ACTIVE_ARCH=NO -derivedDataPath DerivedData -project MeoAsstMac.xcodeproj -scheme MeoAsstMac
+      - uses: actions/upload-artifact@v1
+        with:
+          name: MaaBundle-darwin
+          path: src/MeoAsstMac/DerivedData/Build/Products/Debug
```

Comment:
Consider use [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs)

Reply:
I'm afraid that matrix may not be appropriate at this moment, since the steps do not share with other platforms. On Windows, the workflow uses MSBuild instead of CMake. If the workflow for Linux is implemented in the future, we can adapt this job with matrix. Currently, the job might be like this:

```yaml
  posix-cmake:
    strategy:
      matrix:
        os: [macos-12]
    runs-on: ${{ matrix.os }}
    steps:
        ...
```

Reply:
Makes sense to me :D. Not resolved but dismissed for now. Resolving conversation.